Tasks/XamariniOS/Tests/L0.ts

```diff
@@ -117,9 +117,57 @@ describe('XamariniOS L0 Suite', function () {
         assert(tr.ran('/home/bin/nuget restore src/project.sln'), 'nuget restore should have run');
         assert(tr.ran('/home/bin/msbuild src/project.sln /p:Configuration=Release /p:Platform=iPhone'),
         'msbuild should have run');
-        assert(tr.stderr.length == 0, 'should not have written to stderr');
+        assert(tr.stderr.length === 0, 'should not have written to stderr');
+        assert(tr.succeeded, 'task should have succeeded');
+
+        done();
+    });
+
+    it('XamariniOS can find a single solution file with a glob pattern', function (done: MochaDone) {
+        this.timeout(2000);
+
+        const tp = path.join(__dirname, 'L0SingleWildcardMatch.js');
+        const tr = new ttm.MockTestRunner(tp);
+
+        tr.run();
+
+        assert(tr.stderr.length === 0, 'should not have written to stderr');
+        assert(tr.warningIssues.length === 0, 'should not have issued any warnings');
+        assert(tr.errorIssues.length === 0, 'should not have produced any errors');
+        assert(tr.succeeded, 'task should have succeeded');
+
+        done();
+    });
+
+    it('XamariniOS warns when multiple solution files match a glob pattern', function (done: MochaDone) {
+        this.timeout(2000);
+
+        const tp = path.join(__dirname, 'L0MultipleWildcardMatch.js');
+        const tr = new ttm.MockTestRunner(tp);
+
+        tr.run();
+
+        assert(tr.stderr.length === 0, 'should not have written to stderr');
+        assert(tr.warningIssues.length > 0, 'should have issued a warning');
+        assert(tr.errorIssues.length === 0, 'should not have produced any errors');
```

Comment:
For better validation, you can check the warning message and assert(tr.ran('first solution file matched...'))